### PR TITLE
add nerdctl

### PR DESCRIPTION
**Pull request by neo-liang-sap**
**What this PR does / why we need it**:
add nerdctl per discussion in 

**Which issue(s) this PR fixes**:
Fixes https://github.com/gardener/ops-toolbelt/issues/56

**Special notes for your reviewer**:

**Release note**:

```improvement operator

```

### CODE REVIEW

**Comment by neo-liang-sap** [comment]
local testing results:
image

File: dockerfile-configs/common-components.yaml [hunk 1]
```diff
@@ -63,6 +63,14 @@
   - name: kubetail
     from: https://raw.githubusercontent.com/johanhaleby/kubetail/master/kubetail
     info: Bash script that enables you to aggregate (tail/follow) logs from multiple pods into one stream
+  - name: nerdctl
+    version: v0.9.0
+    from: https://github.com/containerd/nerdctl/releases/download/{version}/nerdctl-full-0.9.0-linux-amd64.tar.gz
+    to: /nerdctl-full-0.9.0-linux-amd64.tar.gz
+    command: |
+      tar Cxzvvf /usr/local nerdctl-full-0.9.0-linux-amd64.tar.gz &&\
+      rm -rf nerdctl-full-0.9.0-linux-amd64.tar.gz
```
Comment (petersutter) [inline dockerfile-configs/common-components.yaml:72]:
```suggestion
      rm -f nerdctl-full-0.9.0-linux-amd64.tar.gz
```
recursive flag not needed in this case

Comment (petersutter) [reply dockerfile-configs/common-components.yaml:72]: is it possible to replace all the other hardcoded version occurrences with the version variable?

Comment (neo-liang-sap) [reply dockerfile-configs/common-components.yaml]: @petersutter it's doable however it requires some relative more work to change sth e.g. add a 'short_version' field in https://github.com/gardener/ops-toolbelt/blob/master/generator/lib/components.py#L93 and change other code which parse curl command , i haven't look into deep yet.
If you consider it's worth doing i will create another ticket to implement this.

meanwhile i see some similar usage e.g. https://github.com/gardener/ops-toolbelt/blob/master/dockerfile-configs/common-components.yaml#L55

what do you think?

Comment (petersutter) [reply dockerfile-configs/common-components.yaml]: ok this is probably too much effort, at least out of scope for this PR. Please ignore my previous comment

Comment (neo-liang-sap) [reply dockerfile-configs/common-components.yaml]: thanks @petersutter

**Review by plkokanov — CHANGES_REQUESTED** [review]
I don't think we need to use the full version of nerdctl (with `full` in the download name) since we would just need the CLI access. Also it is 173 MBs which is too big for the ops image.

**Comment by petersutter** [comment]
> I don't think we need to use the full version of nerdctl (with full in the download name) since we would just need the CLI access. Also it is 173 MBs which is too big for the ops image.

Good point, 173 MB is way too much.  
Btw, meanwhile there is a `v0.10.0` release.

**Comment by neo-liang-sap** [comment]
thanks, i've fixed per comments and testing looks like below
image

**Comment by neo-liang-sap** [comment]
seems there are some network failure in CI building phase
```
Err:1 http://archive.ubuntu.com/ubuntu bionic InRelease
  Temporary failure resolving 'archive.ubuntu.com'
Err:2 http://security.ubuntu.com/ubuntu bionic-security InRelease
  Temporary failure resolving 'security.ubuntu.com'
Err:3 http://archive.ubuntu.com/ubuntu bionic-updates InRelease
  Temporary failure resolving 'archive.ubuntu.com'
Err:4 http://archive.ubuntu.com/ubuntu bionic-backports InRelease
  Temporary failure resolving 'archive.ubuntu.com'
Reading package lists... Done
```

**Review by plkokanov — APPROVED** [review]

**Pull request merged**